src/puppetlabs/kitchensink/core.clj

```diff
@@ -944,6 +944,11 @@ to be a zipper."
   []
   (str (java.util.UUID/randomUUID)))
 
+(defn uuid?
+  "Verifies whether a string is a valid UUID"
+  [uuid]
+  (re-matches #"^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$" uuid))
```

Comment:
I liked the speed of a regex, but fromString would be simpler. And since I primarily planned to use it for schemas and in production we should only check it once, it's probably not too bad.